### PR TITLE
feat: support producer batching configuration

### PR DESCRIPTION
**Pull request by freeznet**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: controllers/spec/utils.go [hunk 1]
```diff
@@ -758,6 +760,27 @@ func generateSourceOutputSpec(source *v1alpha1.Source) *proto.SinkSpec {
...
+		return nil
+	}
+	batchingMaxPublishDelayMs := config.BatchingMaxPublishDelayMs
+	if batchingMaxPublishDelayMs == 0 {
+		// Besides preserving Pulsar's historical default, carrying a non-zero field keeps an explicit
+		// enabled=false distinguishable from an empty BatchingSpec in runtimes using proto3.
+		batchingMaxPublishDelayMs = 10
+	}
+
+	return &proto.BatchingSpec{
```
Comment (jiangpengcheng) [inline controllers/spec/utils.go:774]: [P1] Do not emit `batchingSpec` without defining a compatible runner baseline.

The repository builds and tests runners from `streamnative/sn-platform:3.2.2.1`, whose `Function.proto` does not contain `ProducerSpec.batchingSpec`. Its Java and Python runners reject unknown JSON fields while parsing `--function_details`, so setting `batchingConfig` makes the pod exit before user code starts. The added tests only validate the operator-side proto and cannot catch this incompatibility.

Please add an E2E test using a PIP-401-capable runner and document/enforce the minimum runner version—upstream Pulsar 4.1+—or update the supported runner baseline.

**Review by jiangpengcheng — CHANGES_REQUESTED** [review]
One runner-compatibility issue must be addressed before merge. See the inline finding.

**Review by jiangpengcheng — APPROVED** [review]
